### PR TITLE
Add user-assigned-identity to scripts

### PR DESCRIPTION
**Pull request by sigalsax**
#### What does this PR do?
Add user-assigned-identity information to test scripts and host policy to account for user-assigned-identity workflow

### CODE REVIEW

File: ci/authn-azure/policies/azure-hosts.yml [hunk 1]
```diff
@@ -7,6 +7,13 @@
     - &hosts
       - !host
         id: test-app
```
Comment (orenbm) [inline ci/authn-azure/policies/azure-hosts.yml]: please change to `system-assigned-identity-app`

File: ci/authn-azure/policies/azure-hosts.yml [hunk 2]
```diff
@@ -7,6 +7,13 @@
...
+        annotations:
+          authn-azure/subscription-id: test-subscription
+          authn-azure/resource-group: test-group
+          authn-azure/system-assigned-identity: test-app-pipeline
+
+      - !host
+        id: user-app
         annotations:
           authn-azure/subscription-id: test-subscription
           authn-azure/resource-group: test-group
```
Comment (orenbm) [inline ci/authn-azure/policies/azure-hosts.yml]: please change to `user-assigned-identity-app`

Comment (orenbm) [reply ci/authn-azure/policies/azure-hosts.yml]: also please add an annotation for the user-assigned-identity-app

File: ci/authn-azure/run-authn-azure.sh [hunk 3]
```diff
@@ -1,23 +1,44 @@
...
-# Retrieve a Conjur secret using the authn-azure Conjur access token
-secret=$(curl -k -H "Authorization: Token token=\"$authn_azure_access_token\"" \
-  https://"$CONJUR_SERVER_DNS":8443/secrets/cucumber/variable/secrets/test-variable)
-
-echo "Retrieved secret ${secret} from Conjur!!!"
+function main() {
+    # configure client_id of user identity for Azure instance
+    client_id=""
+
+    user_endpoint="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&client_id=$client_id&resource=https%3A%2F%2Fmanagement.azure.com%2F"
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: `user_endpoint` is not clear. Change to `user_assigned_identity_token_endpoint`

File: ci/authn-azure/run-authn-azure.sh [hunk 4]
```diff
@@ -1,23 +1,44 @@
...
+    conjur_host_user_identity="user-app"
+
+    system_endpoint="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F"
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: `system_endpoint` is not clear. Change to `system_assigned_identity_token_endpoint`

File: ci/authn-azure/run-authn-azure.sh [hunk 5]
```diff
@@ -1,23 +1,44 @@
...
-secret=$(curl -k -H "Authorization: Token token=\"$authn_azure_access_token\"" \
-  https://"$CONJUR_SERVER_DNS":8443/secrets/cucumber/variable/secrets/test-variable)
-
-echo "Retrieved secret ${secret} from Conjur!!!"
+function main() {
+    # configure client_id of user identity for Azure instance
+    client_id=""
+
+    user_endpoint="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&client_id=$client_id&resource=https%3A%2F%2Fmanagement.azure.com%2F"
+    conjur_host_user_identity="user-app"
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: `user_assigned_identity_host_name`

File: ci/authn-azure/run-authn-azure.sh [hunk 6]
```diff
@@ -1,23 +1,44 @@
...
+
+    system_endpoint="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F"
+    conjur_host_system_identity="test-app"
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: `system_assigned_identity_host_name`

File: ci/authn-azure/run-authn-azure.sh [hunk 7]
```diff
@@ -1,23 +1,44 @@
...
+
+    getConjurSecret $user_endpoint $conjur_host_user_identity
+    getConjurSecret $system_endpoint $conjur_host_system_identity
+}
+
+function getConjurSecret() {
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: Please split this function into smaller ones for readability. Also - `getConjurSecret` doesn't really tell the story of what we are doing. Yes, we retrieve a secret but what really is happening here is that we are getting a Conjur access token with an Azure token. Please make it look like it when reading the script.

File: ci/authn-azure/run-authn-azure.sh [hunk 8]
```diff
@@ -1,23 +1,44 @@
...
+    azure_token_endpoint="$1"
+    conjur_role="$2"
+    echo $azure_token_endpoint
+    echo "Retrieving Azure access token"
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: change to `echo "Retrieving Azure access token from $azure_token_endpoint"

File: ci/authn-azure/run-authn-azure.sh [hunk 9]
```diff
@@ -1,23 +1,44 @@
...
+    # Get an Azure access token
+    azure_access_token=$(curl \
+      "$azure_token_endpoint" \
+      -H Metadata:true -s | jq -r '.access_token')
+
+    echo "Get Conjur access token using an Azure access token"
+    # Get an authn-azure Conjur access token for host azure-apps/test-app and user-app
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: Get a Conjur access.

File: ci/authn-azure/run-authn-azure.sh [hunk 10]
```diff
@@ -1,23 +1,44 @@
...
+    azure_token_endpoint="$1"
+    conjur_role="$2"
+    echo $azure_token_endpoint
+    echo "Retrieving Azure access token"
+    # Get an Azure access token
+    azure_access_token=$(curl \
+      "$azure_token_endpoint" \
+      -H Metadata:true -s | jq -r '.access_token')
+
+    echo "Get Conjur access token using an Azure access token"
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: `echo "Get Conjur access token for host $conjur_role using an Azure access token"`

File: ci/authn-azure/run-authn-azure.sh [hunk 11]
```diff
@@ -1,23 +1,44 @@
...
+
+    echo "Retrieve a secret using the Conjur access token"
+    # Retrieve a Conjur secret using the authn-azure Conjur access token
+    secret=$(curl -k -H "Authorization: Token token=\"$authn_azure_access_token\"" \
+      https://"$CONJUR_SERVER_DNS":8443/secrets/cucumber/variable/secrets/test-variable)
+
+    echo "Retrieved secret ${secret} from Conjur!!!"
+}
+
+main
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: newline

File: ci/authn-azure/policies/azure-hosts.yml (hunk 2 above)
Comment (sigalsax) [reply ci/authn-azure/policies/azure-hosts.yml]: Updating `user-assigned-identity-app`

The user-assigned-identity annotation appears in the actual file if you open it

File: ci/authn-azure/policies/azure-hosts.yml [hunk 12]
```diff
@@ -6,7 +6,14 @@
 
     - &hosts
       - !host
-        id: test-app
+        id: system-assigned-identity-app
+        annotations:
+          authn-azure/subscription-id: test-subscription
+          authn-azure/resource-group: test-group
+          authn-azure/system-assigned-identity: test-app-pipeline
```
Comment (orenbm) [inline ci/authn-azure/policies/azure-hosts.yml]: Change (in both hosts) the values to be inside `<>` as they are not real data.

File: ci/authn-azure/run-authn-azure.sh [hunk 13]
```diff
@@ -1,23 +1,64 @@
...
-echo "Retrieved secret ${secret} from Conjur!!!"
+function main() {
+    # configure client_id of user identity for Azure instance
+    client_id=""
+
+    user_assigned_identity_token_endpoint="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&client_id=$client_id&resource=https%3A%2F%2Fmanagement.azure.com%2F"
+    user_assigned_identity_host_name="user-assigned-identity-app"
+
+    system_assigned_identity_token_endpoint="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F"
+    system_assigned_identity_host_name="system-assigned-identity-app"
```
Comment (orenbm) [inline ci/authn-azure/run-authn-azure.sh]: move these lines to be before line 20 (`# use Azure system-assigned-identity to get Conjur access token`)

Comment (sigalsax) [reply ci/authn-azure/run-authn-azure.sh]: good call!

**Pull request merged**